### DISCUSSION TITLE
NoUISlider

### DISCUSSION THREAD

**Discussion by frankvp11** [zauberzeug/nicegui#3205] [Ideas / Feature Requests]
I was wondering if it would be productive to add this element: https://refreshless.com/nouislider/
The reason that I say this is because it allows us to advance beyond the regular range and slider. These two elements are limited in the amount of handles that you can have on the element itself, meaning that you can't have 3 sliders as they do in this example: https://refreshless.com/nouislider/examples/#section-colored-connect
This component also has just more features in general than the components we have now and it would be a good addition (if not replacement) for what we already have.
I've got this mock example set up, if anyone wanted to use this personally. I'd be happy into looking into how we can convert this into an actual component of course. Besides the loading of files, I think it shouldn't be too hard.

main.py:
```python3
from nicegui import ui, element, app

ui.add_body_html("<script src='https://cdn.jsdelivr.net/npm/nouislider/distribute/nouislider.min.js'></script>")
ui.add_body_html("<link rel='stylesheet' href='https://cdn.jsdelivr.net/npm/nouislider/distribute/nouislider.min.css'>")

class NoUISlider(element.Element, component='nouislider.js'):
    def __init__(self):
        super().__init__()
        self.on("slider:update", self._on_update)
    
    def _on_update(self, value):
        print(value)

ui.add_css("""
        #slider {
         width: 80%;
         margin: 10px;
        }
         
         """)

NoUISlider().style("width: 80%")


ui.run()
```

nouislider.js:
```javascript
export default {
    template: `
        <div id='slider' ref='slider'>
        </div>
        <p> Values: <span id="slider-values"></span></p>
    `,
    props: {
        values: {
            type: Array,
            required: false
        }
    },
    mounted() {
        console.log('mounted');
        var slider = document.getElementById('slider');
        console.log(slider);
        noUiSlider.create(slider, {
            start: [20, 40, 60], // Initial positions of handles
            connect: [true, true, true, true],
            range: {
                'min': 0,
                'max': 100
            }
        });

        // Update the display of slider values
        var valuesDisplay = document.getElementById('slider-values');
        slider.noUiSlider.on('update', (values) => {
            console.log(values);
            this.$emit('slider:update', {
                values: values
            });
            valuesDisplay.innerHTML = values.join(', ');
        });
    }
}
```

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @frankvp11!

I've never seen sliders with more than two handles, so I'm not sure how useful it is. But I can imagine adding a `ui.nouislider` element. The integration shouldn't be too complicated.
